client/verta/verta/_dataset.py

```diff
@@ -619,13 +625,49 @@ def make_create_message(dataset_id, dataset_type,
                             version=None):
         raise NotImplementedError('this function must be implemented by subclasses')
 
+    @property
+    def base_path(self):
+        components = self.list_components()
+        base_paths = set(component.base_path for component in components)
+
+        if len(base_paths) == 1:
+            return base_paths.pop()
+        else:  # shouldn't happen
```

Comment:
```suggestion
        else:  # the base path is not defined if they differ for the components
```
"Shouldn't happen" makes it seem like a bug, which it isn't.

Reply:
It is a bug, though. As far as I know, there's no pathway through the Client to end up with different base paths for DatasetVersion components (unlike dataset blobs).

I can clarify this in the comment.

Reply:
```suggestion
        else:  # shouldn't happen: DVs don't have an interface to have different base paths
```